packages/internal/bridge/sdk/src/tokenBridge.ts

```diff
@@ -979,12 +979,86 @@ export class TokenBridge {
  */
   public async getFlowRateWithdrawTx(req:FlowRateWithdrawRequest): Promise<FlowRateWithdrawResponse> {
     // eslint-disable-next-line no-console
-    console.log('stubbed response with req', req);
+
+    if (!req.recipient) {
+      throw new BridgeError(
+        `invalid recipient ${req.recipient}`,
+        BridgeErrorType.INVALID_RECIPIENT,
+      );
+    }
+
+    if (req.index < 0) {
+      throw new BridgeError(
+        `invalid index ${req.index}`,
+        BridgeErrorType.INVALID_FLOWRATE_INDEX,
+      );
+    }
+
+    const rootBridge = await withBridgeError<ethers.Contract>(
+      async () => {
+        const contract = new ethers.Contract(
+          this.config.bridgeContracts.rootERC20BridgeFlowRate,
+          ROOT_ERC20_BRIDGE_FLOW_RATE,
+          this.config.rootProvider,
+        );
+        return contract;
+      },
+      BridgeErrorType.INTERNAL_ERROR,
+    );
+
+    const pending:Array<RootBridgePendingWithdrawal> = await rootBridge.getPendingWithdrawals(req.recipient, [0]);
```

Comment:
I guess this only supports single withdrawal. 
- Is there a reason we don't take the index of the withdrawal as an argument? 
- Can I confirm that there will be a subsequent PR to support multiple withdrawals, or is that not in scope just yet?

Reply:
`rootBridge.getPendingWithdrawals` just gets all the pending withdrawals for an address starting at the index that's passed in. We want to fetch all of the pending withdrawals so we pass in `[0]` to this contract call. we then check there are pending withdrawals and that the `req.index` exists before trying to generate the unsigned transaction for the requested withdrawal index a bit further down.

There is the capability on the contract to sign a TX to execute all pending withdrawals but this is not required by Checkout yet so it's not in scope for this ticket.

Reply:
> starting at the index that's passed in.

are you sure? my recollection was that it takes in an array of indices which are the offsets in the withdrawal array to fetch? so just passing an array with a single element [0] just fetches the first element?

Reply:
let me double check but pretty sure:

https://github.com/immutable/zkevm-bridge-contracts/blob/main/src/root/flowrate/FlowRateWithdrawalQueue.sol#L154

Reply:
hmm actually it looks like you are right now i read the code, from the comment i thought it was just a starting offset. okay ill have to change how this works a bit.

Reply:
i've made the changes, so we now query for the pendingWithdrawalsLength then create an array of indices based on the length and fetch those. I have actually made these changes in the other PR though since the same issue existed over there as well.